enhancements/operator-framework-api/catalogue-check.md

```diff
@@ -0,0 +1,180 @@
+---
+title: neat-enhancement-idea
+authors:
+  - "@camilamacedo86"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-02-19
+last-updated: 2021-02-19
+status: implementable
+see-also:
+  - "/enhancements/olm/max-openshift-versions-for-operators.md"
+---
+
+# OpenshiftCatalogueValidator 
+
+This proposal describes a new set/suite of checks to be implemented in the [operator-framework/api][oper-api] to validate the bundle operator data regarding specific rules for the OpenShift Catalogue.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Currently, [operator-framework/api][oper-api] centralized the code implementation for superset/suites of checks used in the tools and across the board. See [here](https://github.com/operator-framework/api/blob/master/pkg/validation/validation.go#L38-L46) such as:
+
+- `BundleValidator` check if the operator bundle spec is accordingly to spec definition. For further information see [here](https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md).
+- `OperatorHubValidator` checks of the operator is respecting the required criteria to be published in the [OperatorHub.io](https://operatorhub.io/).
+
+The new `OpenShiftCatalogueValidator` proposed here has the purpose of perform checks which are specifically to publish the operator bundle on the OpenShift Catalogue. The validator will raise issues or warnings accordingly.
+
+This checks will also be available to be used by the users via the tools such as SDK and consequently also called in the pipeline. Note that SDK provides currently:
+
+```sh
+To validate a bundle with a superset of requirements for operatorhub.io specifically:
```

Comment:
The superset are the kind of validators. See that if we run only `operator-sdk bundle validate` it will use only the BundleValidator implemented in the operator-framework/api. However, if we inform in the select-option other "superset" such as OperatorHubValidator then it will run its checks. I will try to clarify that before in the EP.

Reply:
It is the SDK output the validators are called/describe as "superset". (it is the current tool output from the help) and then, see: 

- operator-sdk bundle validate -> calls the BundleValidator and ensure that the handle spec def are OK
- operator-sdk bundle validate ./bundle --select-optional name=operatorhub -> calls the OperatorHubValidator to check if the bundle is passing in the criteria to publish it in OperatorHub

Reply:
ok i think i understand.  it might be clearer if this was described as `to perform validation against operatorhub.io requirements, in addition to the standard bundle validation:`

Reply:
It has a full description that might clarify better. See: 

```
 $ operator-sdk bundle validate -h
The 'operator-sdk bundle validate' command can validate both content and format of an operator bundle
image or an operator bundle directory on-disk containing operator metadata and manifests. This command will exit
with an exit code of 1 if any validation errors arise, and 0 if only warnings arise or all validators pass.

A valid bundle is defined by the bundle spec (linked below), therefore the default validator ensures a bundle conforms to	
that spec. If you want to ensure that your bundle is valid for an optional superset of requirements such as to those
required to publish your operator on operatorhub.io, then you will need to run one or more supported optional validators.
Set '--list-optional' to list which optional validators are supported, and how they are grouped by label.
	
More information about operator bundles and metadata:
https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md

NOTE: if validating an image, the image must exist in a remote registry, not just locally.

Usage:
  operator-sdk bundle validate [flags]

Examples:
This example assumes you either have a *pullable* bundle image,
or something similar to the following operator bundle layout present locally:

  $ tree ./bundle
  ./bundle
  ├── manifests
  │   ├── cache.my.domain_memcacheds.yaml
  │   └── memcached-operator.clusterserviceversion.yaml
  └── metadata
      └── annotations.yaml

To validate a local bundle:

  $ operator-sdk bundle validate ./bundle

To build and validate a *pullable* bundle image:

  $ operator-sdk bundle validate <some-registry>/<operator-bundle-name>:<tag>

To list and run optional validators, which are specified by a label selector:

  $ operator-sdk bundle validate --list-optional
  NAME           LABELS                     DESCRIPTION
  operatorhub    name=operatorhub           OperatorHub.io metadata validation
                 suite=operatorframework

To validate a bundle with a superset of requirements for Operator Framework:
	
  $ operator-sdk bundle validate ./bundle --select-optional suite=operatorframework

To validate a bundle with a superset of requirements for operatorhub.io specifically:
	
  $ operator-sdk bundle validate ./bundle --select-optional name=operatorhub


Flags:
  -h, --help                     help for validate
  -b, --image-builder string     Tool to pull and unpack bundle images. Only used when validating a bundle image. One of: [docker, podman, none] (default "docker")
      --list-optional            List all optional validators available. When set, no validators will be run
      --select-optional string   Label selector to select optional validators to run. Run this command with '--list-optional' to list available optional validators

Global Flags:
      --plugins strings          plugin keys of the plugin to initialize the project with
      --project-version string   project version
      --verbose                  Enable verbose logging
```

However, I think that the best approach would be to raise an issue against the SDK repo with we decide to improve it. I will try to add a better context to the EP as well. Please, feel free to ping me for we see how we can make it better in SDK. I will happy in try to work on that.

Reply:
yes, thanks for showing where the term was coming from.  I agree that the SDK help text could be improved, along the lines i suggested.  I don't think `superset` is being used correctly here.

Reply:
Done: https://github.com/operator-framework/operator-sdk/issues/4577